### ISSUE TITLE
"Line contains inconsistent indentation" being reported when not supposed to

### ISSUE THREAD

**Issue by MetaMemoryT** [clutchski/coffeelint#384]
in this code:

```
      @openError or
        @openError = (e) ->
            console.log e.message
            return
```

I think a better version of this code is

```
      unless @openError
        @openError = (e) ->
            console.log e.message
            return
```

So it may be a not that important edge case.

**Comment by AsaAyers** [comment]
indentation is (more or less) based on blocks of code. Your `unless` creates a new block which means it's code should be indented 1 more level. (from CoffeeLint's perspective) In your first example your first two lines belong at the same indentation level because they are part of the same block of code, and your function should be indented 1 level, but is indented twice.

This is an edge case that I don't think needs to be "fixed". I do this kind of thing in my code sometimes and I find the [existential operator](http://coffeescript.org/#operators) to be the best answer here.

    @openError ?= (e) ->
        console.log(e.message)
        return

**Comment by swang** [comment]
This has been discussed previously, basically we can't do much about this because CoffeeScript slurps up the if conditional into one line, thus creating an extra indentation in the next line. Our recommendation has always been to double indent the split conditional line like so:
```cs
@openError or
    @openError = (e) ->
  console.log e.message
  return
```

**Comment by kbaltrinic** [comment]
I am not sure if this is the same issue or related but I am getting this error using a ternary operator in the following manner:

    myVar = if myCondition then
        "the true value (typically a rather long expression)"
      else
        "the false value (could also be a rather long expression)"

It wants me to format it like this which I this is more confusing.

    myVar = if myCondition then
      "the true value (typically a rather long expression)"
    else
      "the false value (could also be a rather long expression)"

**Comment by swang** [comment]
This was fixed one way or another. Closing.
